sensu/plugins/check-large-receive-offload.py

```diff
@@ -0,0 +1,54 @@
+#!/usr/bin/env python
+#
+# Checks that large receive offload is disabled
+#
+# Return CRITICAL if large receive offload is enabled
+#
+# Dean Daskalantonakis <ddaskal@us.ibm.com>
+
+import subprocess
+import argparse
+import sys
+import re
+
+STATE_OK = 0
+STATE_WARNING = 1
+STATE_CRITICAL = 2
+CRITICALITY = 'critical'
+
+def switch_on_criticality():
+    if CRITICALITY == 'warning':
+        sys.exit(STATE_WARNING)
+    else:
+        sys.exit(STATE_CRITICAL)
+
+def main():
+    parser = argparse.ArgumentParser()
+    parser.add_argument('-i', '--interface', help='primary interface')
+    parser.add_argument('-z', '--criticality', default='critical')
+    args = parser.parse_args()
+
+    CRITICALITY = args.criticality
+
+    if args.interface is None:
+        parser.print_help()
+        switch_on_criticality()
+
+    devices = re.findall('eth[0-7]', args.interface)
+
+    for eth in devices:
+        cmd = "ethtool -k %s | grep large-receive-offload | grep ' off'" % (eth)
+
+        lro_check = subprocess.Popen(cmd, shell=True, stdout=subprocess.PIPE, stderr=subprocess.PIPE)
```

Comment:
if this fails, the way the code is written it will return a nonzero value, and exit_with_error_status(critical)... I'm going to find a way to isolate errors due to failed ethtool command rather than the grep result.